File: lib/elo_rating/compare_elo.py
from typing import TypedDict, List
newList = [];

class MovieObject(TypedDict):
    title: str
    elo: int

def calculateTotalElo(movieList: List[List[MovieObject]], movieName: str) -> int:
    totalElo = 0
    # traversing through a 2D array
    for x in movieList:
        for y in x:
            if y.title == movieName:
                totalElo += y.elo;

    return totalElo;

def checkAll(movieList):
    for i in range(len(movieList[0])):
        newElo = calculateTotalElo(movieList, movieList[0][i].title)
        movieObj = MovieObject(title=movieList[0][i].title, elo=newElo)
        newList.append(movieObj)
    return newList

File: lib/elo_rating/compare_elo.ts
// Define the shape of the movie object
export type MovieObject = {
    title: string;
    elo: number;
  };
  
  // Initialize the list that will store the results
  let newList: MovieObject[] = [];
  
  /**
   * Calculates the sum of ELOs for a specific movie name across a 2D list of movies.
   * @param movieList - A 2D array (List of Lists) of MovieObjects
   * @param movieName - The title of the movie to search for
   * @returns The total ELO sum
   */
  function calculateTotalElo(movieList: MovieObject[][], movieName: string): number {
    let totalElo = 0;
  
    // Traversing through the 2D array (equivalent to 'for x in movieList')
    for (const row of movieList) {
      // Equivalent to 'for y in x'
      for (const movie of row) {
        if (movie.title === movieName) {
          totalElo += movie.elo;
        }
      }
    }
  
    return totalElo;
  }
  
  /**
   * Iterates through the first list in the 2D array, calculates totals for those movies,
   * and returns a new list of aggregated MovieObjects.
   * @param movieList - A 2D array of MovieObjects
   * @returns A list of MovieObjects with aggregated ELOs
   */
  export function checkAll(movieList: MovieObject[][]): MovieObject[] {
    // Reset list if needed, or just append as per your python script
    // newList = []; 
    
    // Guard clause to prevent errors if list is empty
    if (movieList.length === 0 || movieList[0].length === 0) {
      return [];
    }
  
    // Iterate through the range of the first sub-list (movieList[0])
    for (let i = 0; i < movieList[0].length; i++) {
      const currentTitle = movieList[0][i].title;
      
      // Calculate the new total ELO
      const newElo = calculateTotalElo(movieList, currentTitle);
  
      // Create the new object
      const movieObj: MovieObject = {
        title: currentTitle,
        elo: newElo
      };
  
      // Append to the list
      newList.push(movieObj);
    }
  
    return newList;
  }